dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -3450,6 +3474,53 @@ Sedp::Endpoint::get_builtin_subscriber() const
   return sedp_.spdp_.bit_subscriber();
 }
 
+EntityId_t Sedp::Endpoint::counterpart_entity_id() const
+{
+  EntityId_t rv = repo_id_.entityId;
+  switch (rv.entityKind) {
+  case DCPS::ENTITYKIND_BUILTIN_WRITER_WITH_KEY:
+    rv.entityKind = DCPS::ENTITYKIND_BUILTIN_READER_WITH_KEY;
+    break;
+
+  case DCPS::ENTITYKIND_BUILTIN_WRITER_NO_KEY:
+    rv.entityKind = DCPS::ENTITYKIND_BUILTIN_READER_NO_KEY;
+    break;
+
+  case DCPS::ENTITYKIND_BUILTIN_READER_NO_KEY:
+    rv.entityKind = DCPS::ENTITYKIND_BUILTIN_WRITER_NO_KEY;
+    break;
+
+  case DCPS::ENTITYKIND_BUILTIN_READER_WITH_KEY:
+    rv.entityKind = DCPS::ENTITYKIND_BUILTIN_WRITER_WITH_KEY;
+    break;
+
+  default:
+    if (DCPS::DCPS_debug_level) {
+      ACE_ERROR((LM_ERROR, "(%P|%t) ERROR: Sedp::Endpoint::counterpart_entity_id: "
+        "Unexpected value: %u\n", rv));
```

Comment:
This is failing in safety profile but may have problems on all configurations - `rv` is not of a type that can be passed to `ACE_Log_Msg` varargs logging.